test/__snapshots__/bindings.test.ts.snap

```diff
@@ -241,17 +516,131 @@ public interface Name extends software.amazon.jsii.JsiiSerializable {
             return result;
         }
     }
 }
 "
 `;
+
+exports[`language bindings 4`] = `
+"// 1bc04b5291c26a46d918139138b992d2de976d6851d0893b0476b85bfbdfc6e6
+package myorg
+
+
+type Name struct {
+	First *string \`json:\\"first\\"\`
+	Last *string \`json:\\"last\\"\`
+	Middle *string \`json:\\"middle\\"\`
+	ObjectWithPropertiesAndBooleanAdditionalProperties *NameObjectWithPropertiesAndBooleanAdditionalProperties \`json:\\"objectWithPropertiesAndBooleanAdditionalProperties\\"\`
+	ObjectWithPropertiesAndTypedAdditionalProperties *NameObjectWithPropertiesAndTypedAdditionalProperties \`json:\\"objectWithPropertiesAndTypedAdditionalProperties\\"\`
+	ObjectWithTypedAdditionalPropertiesOnly *map[string]*string \`json:\\"objectWithTypedAdditionalPropertiesOnly\\"\`
+}
+
+type NameObjectWithPropertiesAndBooleanAdditionalProperties struct {
+	Name *string \`json:\\"name\\"\`
+}
+
+type NameObjectWithPropertiesAndTypedAdditionalProperties struct {
+	Name *string \`json:\\"name\\"\`
+}
+
+"
+`;
+
+exports[`language bindings 5`] = `
+"using Amazon.JSII.Runtime.Deputy;
+
+#pragma warning disable CS0672,CS0809,CS1591
+
+namespace myorg
+{
+    #pragma warning disable CS8618
+
+    /// <remarks>
+    /// <strong>Schema</strong>: Name
+    /// </remarks>
+    [JsiiByValue(fqn: \\"1bc04b5291c26a46d918139138b992d2de976d6851d0893b0476b85bfbdfc6e6.Name\\")]
+    public class Name : myorg.IName
+    {
+        /// <remarks>
+        /// <strong>Schema</strong>: Name#first
+        /// </remarks>
+        [JsiiProperty(name: \\"first\\", typeJson: \\"{\\\\\\"primitive\\\\\\":\\\\\\"string\\\\\\"}\\", isOverride: true)]
+        public string First
+        {
+            get;
+            set;
+        }
+
+        /// <remarks>
+        /// <strong>Schema</strong>: Name#last
+        /// </remarks>
+        [JsiiProperty(name: \\"last\\", typeJson: \\"{\\\\\\"primitive\\\\\\":\\\\\\"string\\\\\\"}\\", isOverride: true)]
+        public string Last
+        {
+            get;
+            set;
+        }
+
+        /// <remarks>
+        /// <strong>Schema</strong>: Name#middle
+        /// </remarks>
+        [JsiiOptional]
+        [JsiiProperty(name: \\"middle\\", typeJson: \\"{\\\\\\"primitive\\\\\\":\\\\\\"string\\\\\\"}\\", isOptional: true, isOverride: true)]
+        public string? Middle
+        {
+            get;
+            set;
+        }
+
+        /// <remarks>
+        /// <strong>Schema</strong>: Name#objectWithPropertiesAndBooleanAdditionalProperties
+        /// </remarks>
+        [JsiiOptional]
+        [JsiiProperty(name: \\"objectWithPropertiesAndBooleanAdditionalProperties\\", typeJson: \\"{\\\\\\"fqn\\\\\\":\\\\\\"1bc04b5291c26a46d918139138b992d2de976d6851d0893b0476b85bfbdfc6e6.NameObjectWithPropertiesAndBooleanAdditionalProperties\\\\\\"}\\", isOptional: true, isOverride: true)]
+        public myorg.INameObjectWithPropertiesAndBooleanAdditionalProperties? ObjectWithPropertiesAndBooleanAdditionalProperties
```

Comment:
This should have been somehow an intersection with `System.Collections.Generic.IDictionary<string, System.Object>`